### PR TITLE
Add ability to scale HUD text

### LINKED ISSUE #4898: HUD Text Scale
Adding text to a HUD element is great, even having it's own colour, but we really need the ability to scale the text size and override the default GUI_Scale setting in .conf e.g.
```
local idx = player:hud_add({
	hud_elem_type = "text",
	position = {x = 1, y = 0},
	offset = {x=-100, y = 20},
	scale = {x = 100, y = 100},
	text = "My Text",
	text_size = 2.4,
	number = 0xFF0000 -- Red
})

### PR DESCRIPTION
**Pull request by LoneWolfHT**
Closes #4898

## To do

Ready for Review.

**I will remove the minimal tests when this is ready to be merged**

## How to test

Join the minimal game with different font sizes and look at your HUD

### CODE REVIEW

File: src/client/hud.cpp [hunk 1]
```diff
@@ -322,16 +322,17 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 					NULL, colors, true);
 				break; }
 			case HUD_ELEM_TEXT: {
+				irr::gui::IGUIFont* textfont = (e->size.X > 0) ? g_fontengine->getFont(e->size.X * g_fontengine->getDefaultFontSize()) : font;
```
Comment (sfan5) [inline src/client/hud.cpp:325]: line too long

Comment (LoneWolfHT) [reply src/client/hud.cpp]: How is it now?

**Comment by LoneWolfHT** [comment]
Looks like VSCode removed whitespace at the end of some lines in `lua_api.txt`. Is that ok?

File: src/client/hud.cpp [hunk 2]
```diff
@@ -322,7 +322,10 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 					NULL, colors, true);
 				break; }
 			case HUD_ELEM_TEXT: {
-				irr::gui::IGUIFont* textfont = (e->size.X > 0) ? g_fontengine->getFont(e->size.X * g_fontengine->getDefaultFontSize()) : font;
+				irr::gui::IGUIFont* textfont = font;
+				if (e->size.X > 0)
+					textfont = g_fontengine->getFont(e->size.X * g_fontengine->getDefaultFontSize());
```
Comment (ClobberXD) [inline src/client/hud.cpp:327]: It still seems like this line is pretty long (102 chars, with a tab width of 4) - maybe split this line after `getFont(`?

```suggestion
					textfont = g_fontengine->getFont(
						e->size.X * g_fontengine->getDefaultFontSize());
```

Comment (LoneWolfHT) [reply src/client/hud.cpp:327]: There's a line below a ways that's 101 characters long, so I'm not sure

Comment (LoneWolfHT) [reply src/client/hud.cpp:327]: Eh, why not

File: doc/lua_api.txt [hunk 3]
```diff
@@ -1327,6 +1327,7 @@ Displays text on the HUD.
   text. Specify `0xFFFFFF` for white text, `0xFF0000` for red, and so on.
 * `alignment`: The alignment of the text.
 * `offset`: offset in pixels from position.
+* `size`: size of the text. The player-set font size is multiplied by size.x (y value isn't used).
```
Comment (ClobberXD) [inline doc/lua_api.txt:1330]: Why is this a table, if there's only one value in it? :thinking:

**Comment by ClobberXD** [comment]
> Looks like VSCode removed whitespace at the end of some lines in `lua_api.txt`. Is that ok?

Those whitespaces seem to have been accidentally added in a previous PR. They should be removed. :+1:

File: doc/lua_api.txt (hunk 3 above)
Comment (LoneWolfHT) [reply doc/lua_api.txt:1330]: It's used as a table by other element types

Comment (SmallJoker) [reply doc/lua_api.txt:1330]: Line too long. Please break it.

File: games/minimal/mods/test/hudscaletest.lua [hunk 4]
```diff
@@ -0,0 +1,55 @@
+minetest.register_on_joinplayer(function(player)
+	minetest.after(1, function()
+		player:hud_add({
+			hud_elem_type = "text",
+			position = {x = 0.5, y = 0.4},
+			scale = {x = 100, y = 100},
+			text = "Small Hello World",
+			number = 0xFFFFFF,
+			alignment = {x = 1, y = 0},
+		})
```
Comment (SmallJoker) [inline games/minimal/mods/test/hudscaletest.lua:10]: Two examples should be enough to demonstrate/test this feature.

File: src/client/hud.cpp [hunk 5]
```diff
@@ -322,16 +322,21 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 					NULL, colors, true);
 				break; }
 			case HUD_ELEM_TEXT: {
+				irr::gui::IGUIFont* textfont = font;
```
Comment (SmallJoker) [inline src/client/hud.cpp:325]:
```suggestion
				irr::gui::IGUIFont *textfont = font;
```

File: games/minimal/mods/test/hudscaletest.lua (hunk 4 above)
Comment (LoneWolfHT) [reply games/minimal/mods/test/hudscaletest.lua:10]: How about 3? 
image

File: doc/lua_api.txt (hunk 3 above)
Comment (LoneWolfHT) [reply doc/lua_api.txt]: Is a newline and two spaces the norm for doing that? Done if so

**Comment by oilboi** [comment]
This is insanely useful

File: doc/lua_api.txt (hunk 3 above)
Comment (SmallJoker) [reply doc/lua_api.txt]: Yes exactly. Indent to the first character after the list point.

File: games/minimal/mods/test/hudscaletest.lua (hunk 4 above)
Comment (ClobberXD) [reply games/minimal/mods/test/hudscaletest.lua]: Is the number of examples a concern when they're going to be removed before merge? Or did I miss an update?

****

But wow, the screenshot looks good... :)

Comment (LoneWolfHT) [reply games/minimal/mods/test/hudscaletest.lua]: They're quite intrusive and only cover the one example. I'd prefer to remove them before merge

**Comment by LoneWolfHT** [comment]
Minimal tests were removed

**Pull request merged**